File: String/Q4.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string result="";
        int n=strs.size();
        sort(strs.begin(),strs.end());
        string first=strs[0];
        string last=strs[n-1];
        
        for(int i=0;i<first.size();i++){
            if(first[i]==last[i]){
                result+=first[i];
            }
            else{
                return result;
            }
        }
        return result;
    }
};

File: String/Q4.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        result = ""
        n= len(strs)
        strs.sort()
        
        fi = strs[0]
        se = strs[n-1]
        
        for i in range(0,len(fi)):
            if fi[i] == se[i]:
                result = result+fi[i]
            else:
                return result
        return result
